source/mqtt_agent.c

```diff
@@ -200,6 +200,24 @@ static MQTTStatus_t createAndAddCommand( CommandType_t commandType,
                                          CommandContext_t * pCommandCompleteCallbackContext,
                                          uint32_t blockTimeMs );
 
+/**
+ * @brief Resend QoS 1 and 2 publishes after resuming a session.
+ *
+ * @param[in] pMqttAgentContext Agent context for the MQTT connection.
+ *
+ * @return MQTTSuccess if all publishes resent successfully, else error code
+ * from #MQTT_Publish.
+ */
+static MQTTStatus_t resendPublishes( MQTTAgentContext_t * pMqttAgentContext );
+
+/**
+ * @brief Clear the list of pending acknowledgments by invoking each callback
+ * with #MQTTBadParameter.
```

Comment:
Using MQTTBadParameter strikes me as odd.  Would MQTTRecvFailed be a better choice?

Reply:
Good catch. I actually used MQTTBadResponse but documented it incorrectly. Should I change them both to MQTTRecvFailed or keep as MQTTBadResponse?

Reply:
I don't think there is a best answer.  My first thought for bad response would be "I got something and it was bogus."  The receive failed makes me think "I called socket read and got an error." which is still not quite right, but is a better fit for a lost connection.  What do you think?

Reply:
I think best would be if this situation had its own error code, but given the current implementation decision of using coreMQTT's codes, I agree with you that receive failed is the closest fit.